### PR TITLE
Use Connector.lookupScanCredentials to contextualize GitHubAppCredentials

### PR DESCRIPTION
**Pull request by jeromepochat**
https://github.com/jenkinsci/github-checks-plugin/pull/396 fixed `GitHubAppCredentials` owner inference using `withOwner` method which not considered as a public API. 

This uses `Connector.lookupScanCredentials` from GitHub Branch Source plugin to contextualize the credentials automatically. It removes the direct usage of `GitHubAppCredentials.withOwner` from `github-checks-plugin`.

This includes some mocking adjustments to involve the [credentials contextualization](https://github.com/jenkinsci/github-branch-source-plugin/blob/4ed4bdfc4a6a83384d2f521b666cdc59d3428c45/src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java#L310) provided by `github-branch-plugin`.

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java [hunk 1]
```diff
@@ -159,56 +159,66 @@ public static Collection<Object[]> contextBuilders() {
...
-                        .build())
-                .withActions(Collections.singletonList(
-                        new ChecksAction("re-run", "re-run Jenkins build", "#0")))
-                .build();
-
-        new GitHubChecksPublisher(contextBuilder.apply(this),
-                new PluginLogger(j.createTaskListener().getLogger(), "GitHub Checks"),
-                wireMockRule.baseUrl())
-                .publish(details);
+        try (var credentialsMatchers = mockCredentialsMatchers()) {
```
Comment (jeromepochat) [inline src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java:175]: Introduced `try-with-resource` for static mock auto-close. The body is indented without any change.

File: src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java [hunk 2]
```diff
@@ -217,48 +227,50 @@ public void shouldPublishGitHubCheckRunCorrectly() {
...
-                .contains("title='Jenkins Check'")
-                .contains("summary='# A Successful Build'")
-                .contains("path='Jenkinsfile'")
-                .contains("startLine=1")
-                .contains("endLine=2")
-                .contains("startColumn=0")
-                .contains("endColumn=20")
-                .contains("annotationLevel=WARNING")
-                .contains("message='say hello to Jenkins'");
+        try (var credentialsMatchers = mockCredentialsMatchers()) {
```
Comment (jeromepochat) [inline src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java:230]: Introduced `try-with-resource` for static mock auto-close. The body is indented without any change.

File: src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java [hunk 3]
```diff
@@ -313,8 +325,9 @@ public void testChecksPublisherUpdatesCorrectly() throws Exception {
         when(repository.createCheckRun(eq(checksName2), anyString())).thenReturn(createBuilder2);
         when(repository.updateCheckRun(checksId1)).thenReturn(updateBuilder1);
 
-        try (MockedStatic<Connector> connector = mockStatic(Connector.class)) {
-            connector.when(() -> Connector.connect(anyString(), any(GitHubAppCredentials.class))).thenReturn(gitHub);
+        try (var credentialsMatchers = mockCredentialsMatchers(); var connector = mockStatic(Connector.class)) {
+            connector.when(() -> Connector.lookupScanCredentials(any(), any(), any(), any())).thenCallRealMethod();
```
Comment (jeromepochat) [inline src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java:329]: Call real method to involve GitHubAppCredentials contextualization

File: src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java [hunk 4]
```diff
@@ -435,11 +448,8 @@ GitHubChecksContext createGitHubChecksContextWithGitHubSCM(final J job) throws E
         when(source.getRepoOwner()).thenReturn("XiongKezhi");
         when(source.getRepository()).thenReturn("Sandbox");
 
-        GitHubAppCredentials gitHubAppCredentials = new GitHubAppCredentials(CredentialsScope.GLOBAL,
-                "cred-id", null, "app-id", Secret.fromString(TEST_PRIVATE_KEY));
-
         when(scmFacade.findGitHubSCMSource(job)).thenReturn(Optional.of(source));
-        when(scmFacade.findGitHubAppCredentials(job, "1")).thenReturn(Optional.of(gitHubAppCredentials));
+        when(scmFacade.findGitHubAppCredentials(job, "1")).thenCallRealMethod();
```
Comment (jeromepochat) [inline src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java:452]: Call real method to involve GitHubAppCredentials contextualization

**Review by timja — APPROVED** [review]

**Pull request merged**